.github/workflows/ci.yml

```diff
@@ -26,4 +27,49 @@ jobs:
         working-directory: ./functions/generated/pyfirebase
         run: uv sync --all-extras
 
-      - uses: pre-commit/action@v3.0.1
+      - name: Pre-commit check
+        uses: pre-commit/action@v3.0.1
+
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'yarn'
+          cache-dependency-path: ./functions/yarn.lock
+
+      - name: Install yarn
+        run: npm install -g yarn@1.22.22
```

Comment:
We can use an action for setup yarn
https://github.com/threeal/setup-yarn-action